pkg/client/toolchain_object.go

```diff
@@ -37,19 +35,14 @@ type toolchainObjectImpl struct {
 }
 
 // NewToolchainObject returns an instance of ToolchainObject for the given runtime.Object
-func NewToolchainObject(ob client.Object) (ToolchainObject, error) {
-	if ob == nil {
+func NewToolchainObject(obj client.Object) (ToolchainObject, error) {
+	if obj == nil {
 		return nil, fmt.Errorf("the provided object is nil, so the constructor cannot create an instance of ToolchainObject")
 	}
-	accessor, err := meta.Accessor(ob)
-	if err != nil {
-		return nil, errors.Wrapf(err, "unable to get accessor of object %v", ob)
-	}
-
 	return &toolchainObjectImpl{
-		Object:       accessor,
-		gvk:          ob.GetObjectKind().GroupVersionKind(),
-		clientObject: ob,
+		Object:       obj,
+		gvk:          obj.GetObjectKind().GroupVersionKind(),
+		clientObject: obj,
```

Comment:
What is the purpose of keeping the `Object` as well as `clientObject`? It's the same.

Reply:
it's being simplified in https://github.com/codeready-toolchain/toolchain-common/pull/211 ;)